flagd 0.3.0 (new formula)

josecolella opened the pull request:
Signed-off-by: Jose Colella <jose.colella@gusto.com>

<!-- Use [x] to mark item done, or just click the checkboxes with device pointer -->

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

chenrui333 commented on Formula/flagd.rb:
```diff
@@ -0,0 +1,20 @@
+  end
+
+  test do
+    assert_predicate bin/"flagd", :exist?
+
```
That is unnecessary.
```suggestion
```

josecolella replied:
@chenrui333 Thanks. Removing

chenrui333 commented on Formula/flagd.rb:
```diff
@@ -0,0 +1,20 @@
+  test do
+    assert_predicate bin/"flagd", :exist?
+
+    assert_match(/Flagd is a simple command line tool/, shell_output("#{bin}/flagd"))
```
We need to assert the command usage, not help text.

josecolella replied:
@chenrui333 Since this is a daemon process, do we have any examples or testing services that I can use as reference?

SMillerDev replied:
HTTPD is a daemon process that is tested.

chenrui333 replied:
also as this is daemon process, you can add plist as well

SMillerDev replied:
A service block, not a plist.

chenrui333 replied:
yeah, service block (we deprecated plist) :)

josecolella replied:
Thanks @SMillerDev and @chenrui333 . I've added tests for this

chenrui333 commented on Formula/flagd.rb:
```diff
@@ -0,0 +1,20 @@
+class Flagd < Formula
+  desc "Feature flag daemon with a Unix philosophy"
+  homepage "https://github.com/open-feature/flagd"
+  url "https://github.com/open-feature/flagd.git", using: :git, tag: "v0.3.0", revision: "2ad206e635cc5b68ae42287b740ddfc86b0194ff"
```
```suggestion
  url "https://github.com/open-feature/flagd.git"
      tag: "v0.3.0",
      revision: "2ad206e635cc5b68ae42287b740ddfc86b0194ff"
```

josecolella replied:
@chenrui333 Thanks. I will modify

chenrui333 commented on Formula/flagd.rb:
```diff
@@ -0,0 +1,20 @@
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPRIVATE"] = "buf.build/gen/go"
+    system "go", "build", "-o", bin/"flagd"
```
```suggestion
    ENV["GOPRIVATE"] = "buf.build/gen/go"
    system "go", "build", *std_go_args

    generate_completions_from_executable(bin/"flagd", "completion")
```

josecolella replied:
Thanks.

josecolella commented on Formula/flagd.rb:
```diff
@@ -0,0 +1,45 @@
+
+  def install
+    ENV["GOPRIVATE"] = "buf.build/gen/go"
+    ldflags = %W[
+      -s -w
+      -X main.version=#{version}
+      -X main.commit=#{Utils.git_head}
+      -X main.date=#{time.iso8601}
+    ]
```
@SMillerDev @chenrui333 I added this as well since it was missing from the initial implementation

josecolella commented on Formula/flagd.rb:
```diff
@@ -0,0 +1,45 @@
+      end
+      sleep 3
+
+      resolve_boolean_command = <<-BASH
+        curl -X POST "localhost:#{port}/schema.v1.Service/ResolveBoolean" -d '{"flagKey":"myBoolFlag","context":{}}' -H "Content-Type: application/json"
+      BASH
+
+      expected_output = { "value": true, "reason": 'STATIC', "variant": 'on' }.to_json
+
+      assert_match expected_output, shell_output(resolve_boolean_command)
```
validating that daemon returns feature flags values: https://github.com/open-feature/flagd/blob/main/docs/usage/evaluation_examples.md

SMillerDev commented on Formula/flagd.rb:
```diff
@@ -0,0 +1,46 @@
+class Flagd < Formula
+  desc "Feature flag daemon with a Unix philosophy"
+  homepage "https://github.com/open-feature/flagd"
+  url "https://github.com/open-feature/flagd.git", using: :git, tag: "v0.3.0", revision: "2ad206e635cc5b68ae42287b740ddfc86b0194ff"
```
```suggestion
  url "https://github.com/open-feature/flagd.git", 
  tag: "v0.3.0", 
  revision: "2ad206e635cc5b68ae42287b740ddfc86b0194ff"
```

josecolella replied:
Thanks

SMillerDev commented on Formula/flagd.rb:
```diff
@@ -0,0 +1,46 @@
+      -X main.date=#{time.iso8601}
+    ]
+
+    system "go", "build", *std_go_args(ldflags: ldflags, output: bin/"flagd")
```
That's the default
```suggestion
    system "go", "build", *std_go_args(ldflags: ldflags)
```

josecolella replied:
Thanks

josecolella commented:
@chenrui333 @SMillerDev I've addressed the comments. When you have a moment to take a look again. Thanks again for all the feedback. This has been a great learning opportunity

josecolella commented:
@chenrui333 It mentions that only those with write access can merge. Is this something that is done by maintainers of the project?

p-linnane commented:
@josecolella I've triggered a merge. Thanks for your contribution to Homebrew!

josecolella commented:
@p-linnane Thank you. I've used homebrew extensively as a developer. It feels great to contribute back to the community in some way 🥳
